### ISSUE TITLE
ClassTalentTalentsTabMixin:LoadConfigByIndex does not compare indexes in predicate

### ISSUE THREAD

**Issue by Caedis** [Stanzilla/WoWUIBugs#345]
WoW Version: Beta (10.0.2.46619)

[Code reference](https://github.com/Gethe/wow-ui-source/blob/beta/Interface/AddOns/Blizzard_ClassTalentUI/Blizzard_ClassTalentTalentsTab.lua#L1262)
Shows that the arg passed into the function is not used besides a bounds check.

Issue in game:
Running the new `/loadoutindex` or `/loi` with a known config index gives a red error message (`ERR_TALENT_FAILED_UNKNOWN`/"Unable to learn talent.")

## Solution
Note: the first arg passed into the predicate's function is the index as listed [here](https://github.com/Gethe/wow-ui-source/blob/beta/Interface/AddOns/Blizzard_ClassTalentUI/Blizzard_ClassTalentTalentsTab.lua#L1226)  

Change the predicate used to limit the config to check the `index` arg against the first arg passed to the predicate.

```diff
diff --git a/Interface/AddOns/Blizzard_ClassTalentUI/Blizzard_ClassTalentTalentsTab.lua b/Interface/AddOns/Blizzard_ClassTalentUI/Blizzard_ClassTalentTalentsTab.lua
index 5ec446b9..bef864dc 100644
--- a/Interface/AddOns/Blizzard_ClassTalentUI/Blizzard_ClassTalentTalentsTab.lua
+++ b/Interface/AddOns/Blizzard_ClassTalentUI/Blizzard_ClassTalentTalentsTab.lua
@@ -1259,9 +1259,8 @@ function ClassTalentTalentsTabMixin:LoadConfigByIndex(index)
                return;
        end
 
-       self:LoadConfigByPredicate(function(_, configID)
-               local lowerConfigName = self.configIDToName[configID] and self.configIDToName[configID]:lower();
-               return lowerConfigName == lowerTargetName;
+       self:LoadConfigByPredicate(function(configIndex)
+               return configIndex == index;
        end);
 end
```

**Comment by Caedis** [comment]
also reported in game with a reference to this

**Comment by Caedis** [comment]
Fixed in ptr (10.0.2.46802)
